### PR TITLE
Hatohol house keeper

### PR DESCRIPTION
**Pull request by bynnchapu**
I have created Hatohol house keeper tool. (#779)
But, this tool is not perfect, I think this tool has following TODO items.
I hope @t-miyazawa san fixes following TODO items.

TODO:
- Use ```hatohol.conf``` file
  - This House keeper tool doesn't use it
- Show message about "It takes long time"
  - Cause of long time is storage engine of MySQL (MariaDB).
  - Otherwise, will we add index into ```events``` and ```action_logs``` tables ?

### CODE REVIEW

File: server/tools/hatohol-house-keeper [hunk 1]
```diff
@@ -0,0 +1,113 @@
+#!/usr/bin/env python
+import sys
+import os
+import argparse
+import MySQLdb
+import time
+
+
+def finish_if_require_option_is_fault(args):
```
Comment (kz0817) [inline server/tools/hatohol-house-keeper]: required ?

Comment (kz0817) [inline server/tools/hatohol-house-keeper]: The sentence "option is fault" doesn't makes sense.
Do you mean  "finish_if_required_option_is_missing"

File: server/tools/hatohol-house-keeper [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+    if args.date:
+        delete_unnecessary_records_by_date(args.table, args.date, db, cursor)
+    if args.number:
+        delete_unncessary_records_by_number(args.table, args.number, db, cursor)
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser(description='Hatohol House Keeper')
+    parser.add_argument('-d', '--date', type=str,
+                        help='A term of removed records. Format: YY-mm-dd')
```
Comment (kz0817) [inline server/tools/hatohol-house-keeper]: Options -d and -n are exclusive, don't they.
Can we use sub command parser of argparse ?

Comment (kz0817) [inline server/tools/hatohol-house-keeper]: How about check the time format and creation of time of datetime object here by a custom parser.

File: server/tools/hatohol-house-keeper [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+    except:
+        db.rollback()
+
+
+def delete_unnecessary_records_by_date(table_name, date, db, cursor):
+    if 'events' in table_name:
+        time_sec = return_unix_time(date)
+        query = 'DELETE FROM events WHERE time_sec < %d' % time_sec
+    elif 'action_logs' in table_name:
+        query = 'DELETE FROM action_logs WHERE queuing_time < %s' % date
```
Comment (kz0817) [inline server/tools/hatohol-house-keeper]: Do all the action log records have valid queuing_time ?
This is just a question. I forget the behavior.

Comment (bynnchapu) [inline server/tools/hatohol-house-keeper]: I checked the table of ```action_logs``` table.
It seems all action log records have valid ```queuing_time```.

I want to check the source code, but I can not find correspond source code.
(I'm bussy now, so I don't have time to show correspond source code. Somehow I have time to fix above comment...)

**Comment by bynnchapu** [comment]
I close this pull request because I don't finish this work by code freeze.
I'll fix above comments, re-open or re-send pull request.

**Pull request closed without merging**